Review Enrollment API

stvstnfrd opened the pull request.

stvstnfrd committed:
SQUASH: Cleanup include

stvstnfrd committed:
SQUASH: Fix auto_enroll param?

stvstnfrd committed:
SQUASH: Whitespace and formatting

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/views.py:
```diff
@@ -55,38 +55,40 @@ def post(self, request):
         email_students = request.POST.get('email_students', False) in ['true', 'True', True]
-        auto_enroll = request.POST.get('email_students', False) in ['true', 'True', True]
-
+        auto_enroll = request.POST.get('auto_enroll', False) in ['true', 'True', True]
```
Is this the proper fix?

kluo replied:
oh woops, I must not have saved my local change for this somehow, thanks for the catch!

stvstnfrd commented on common/djangoapps/enrollment/urls.py:
```diff
@@ -3,7 +3,8 @@
 """
 from django.conf import settings
-from django.conf.urls import patterns, include, url
+from django.conf.urls import include
```
We're less likely to get conflicts in the future if we break it out onto its own line.

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/views.py:
```diff
@@ -55,38 +55,40 @@ def post(self, request):
             return Response(
                 status=status.HTTP_400_BAD_REQUEST,
-                data={"message": u"Invalid or missing course_id"},
+                data={
```
Just went ahead and reformatted some dicts and non-interpolated strings :)

stvstnfrd commented:
@kluo

stvstnfrd committed:
SQUASH: Try update endpoint URL

The URL doesn't need to mention the action (update),
that is implied via the HTTP method (POST).

Instead, we can abstract this a bit and use it again in the future.
The other fork we had of `get_roster` could be expressed as a GET of
this same URL here (get the enrollment status of a course).

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/urls.py:
```diff
@@ -3,13 +3,13 @@
     url(
-        r'^update_status$',
-        UpdateEnrollmentView.as_view(),
+        r'^enrollment/status/{course_key}/?$',
```
Optional trailing slash.

stvstnfrd replied:
Oh whoops, or is this a `course_id` and the `course_key` is the Python object?

stvstnfrd replied:
FIXED NOW: Whoops, also I need to format the course_key regexp into that..

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/views.py:
```diff
@@ -35,23 +35,23 @@ class UpdateEnrollmentView(APIView):
     @method_decorator(require_post_params(['email', 'course_id', 'action']))
     @method_decorator(ensure_csrf_cookie_cross_domain)
-    def post(self, request):
+    def post(self, request, course_key):
```
This is the correct, syntax, right?

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/views.py:
```diff
@@ -35,23 +35,23 @@ class UpdateEnrollmentView(APIView):
         **Example Request**
 
-            POST /api/enrollment/v1/update_status
+            POST /api/enrollment/v1/status/course-v1:foo+bar+foobar
```
Is this what it would look like or no? I'm a bit confused about how this URL gets constructed.

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/views.py:
```diff
@@ -35,23 +35,23 @@ class UpdateEnrollmentView(APIView):
+            POST /api/enrollment/v1/status/course-v1:foo+bar+foobar
             {
                 'email': 'foo@bar.com',
-                'course_id': 'course-v1:foo+bar+foobar',
```
This comes directly from the URL now.

stvstnfrd committed:
SQUASH-typo

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/views.py:
```diff
@@ -35,23 +35,23 @@ class UpdateEnrollmentView(APIView):
         """
         try:
-            course_id = CourseKey.from_string(request.data['course_id'])
+            course_key = request.data['course_key']
```
Don't need this line. Deleted in next commit.

stvstnfrd committed:
SQUASH: Fix url

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/urls.py:
```diff
@@ -1,15 +1,18 @@
     url(
-        r'^update_status$',
-        UpdateEnrollmentView.as_view(),
+        r"^enrollment/status/{course_key}/?$".format(
```
Is the `status` namespace enough? Or do we need to explicitly put it behind `stanford/` something or other?

stvstnfrd replied:
I just want to make sure we don't step on edx's toes.. or get ours stepped on..
If we do something generic enough maybe it'll be easier to land upstream? 🤷‍♂️

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/urls.py:
```diff
@@ -1,15 +1,18 @@
 urlpatterns = patterns(
     'enrollment.views',
     url(
-        r'^update_status$',
```
The URL doesn't need to mention the action (update),
that is implied via the HTTP method (POST).
    
Instead, we can abstract this a bit and use it again in the future.
The other fork we had of `get_roster` could be expressed as a GET of
this same URL here (get the enrollment status of a course).

stvstnfrd replied:
I also foresee us wanting to lookup the status of a particular user, which could also be done w/ a GET of this URL, accompanied by the `email` parameter.

stvstnfrd committed:
Add init

stvstnfrd committed:
SQUASH: fix course id/key

stvstnfrd committed:
SQUASH: fix course id/key

stvstnfrd committed:
SQUASH: put course_id back in payload, out of URL

stvstnfrd committed:
SQUASH: put course_id back in URL

stvstnfrd committed:
Revert "SQUASH: put course_id back in URL"

This reverts commit 5dadce5793f840645f87a84327e44097df98cd34.

stvstnfrd committed:
SQUASH: move

stvstnfrd commented on openedx/stanford/common/djangoapps/enrollment/views.py:
```diff
@@ -51,54 +51,59 @@ def post(self, request):
         email_students = request.POST.get('email_students', False) in ['true', 'True', True]
-        auto_enroll = request.POST.get('email_students', False) in ['true', 'True', True]
-
+        auto_enroll = request.POST.get('auto_enroll', False) in ['true', 'True', True]
```
@kluo Do we actually care to be this forgiving? Shouldn't we just enforce one style?

kluo replied:
Well I was maintaining how it currently checks from instructor dashboard:
https://github.com/Stanford-Online/edx-platform/blob/master/lms/djangoapps/instructor/views/api.py#L3349
